New swapchain model

kvark opened the pull request:
Fixes #322
TODO:
- [x] Test on Metal
- [x] Test on DX12
- [x] Test on Vulkan
- [x] Try embedding the backend information into `SwapChainId`
- [x] Wait for https://github.com/amethyst/rendy/pull/202

kvark commented:
@grovesNL this is all functional and ready for the review!

grovesNL commented on wgpu-native/src/swap_chain.rs:
```diff
@@ -133,221 +114,120 @@ impl SwapChainDescriptor {
 #[derive(Debug)]
 pub struct SwapChainOutput {
-    pub texture_id: TextureId,
+    //pub texture_id: TextureId,
```
nit: leftover comment

grovesNL commented on wgpu-native/src/swap_chain.rs:
```diff
@@ -1,81 +1,62 @@
+
+    ## Lifecycle
+
+    At the low level, the swap chain is using the new simplified model of gfx-rs.
```
Thanks for the explanation, this is really helpful!

kvark commented:
@grovesNL thanks for the review!
I rebased the code and updated the description of what's actually done.
Note that the biggest unknown here is whether just having a `TextureView` for the swapchain image is enough... I'll be talking about this on the upcoming F2F. If that's not the case, we'd have to go back to gfx-rs and extend its new swapchain model, which is not trivial.

Anyhow, let's proceed 🚀 
bors r=grovesNL
